internal/internal_task_pollers_test.go

```diff
@@ -54,3 +68,132 @@ func TestLocalActivityPanic(t *testing.T) {
 }
+
+func TestRespondTaskCompleted_failed(t *testing.T) {
+	t.Run("fail sends RespondDecisionTaskFailedRequest", func(t *testing.T) {
+		testTaskToken := []byte("test-task-token")
+
+		poller, client, _, _ := buildWorkflowTaskPoller(t)
+		client.EXPECT().RespondDecisionTaskFailed(gomock.Any(), &s.RespondDecisionTaskFailedRequest{
+			TaskToken:      testTaskToken,
+			Cause:          s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure.Ptr(),
+			Details:        []byte(assert.AnError.Error()),
+			Identity:       common.StringPtr(_testIdentity),
+			BinaryChecksum: common.StringPtr(getBinaryChecksum()),
+		}, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+
+		res, err := poller.RespondTaskCompleted(nil, assert.AnError, &s.PollForDecisionTaskResponse{
+			TaskToken: testTaskToken,
+			Attempt:   common.Int64Ptr(0),
+		}, time.Now())
+		assert.NoError(t, err)
+		assert.Nil(t, res)
+	})
+	t.Run("fail fails to send RespondDecisionTaskFailedRequest", func(t *testing.T) {
+		testTaskToken := []byte("test-task-token")
+
+		poller, client, _, _ := buildWorkflowTaskPoller(t)
+		client.EXPECT().RespondDecisionTaskFailed(gomock.Any(), &s.RespondDecisionTaskFailedRequest{
+			TaskToken:      testTaskToken,
+			Cause:          s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure.Ptr(),
+			Details:        []byte(assert.AnError.Error()),
+			Identity:       common.StringPtr(_testIdentity),
+			BinaryChecksum: common.StringPtr(getBinaryChecksum()),
+		}, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(assert.AnError)
+
+		// We cannot test RespondTaskCompleted since it uses backoff and has a hardcoded retry mechanism for 60 seconds.
+		_, err := poller.respondTaskCompletedAttempt(errorToFailDecisionTask(testTaskToken, assert.AnError, _testIdentity), &s.PollForDecisionTaskResponse{
+			TaskToken: testTaskToken,
+			Attempt:   common.Int64Ptr(0),
+		})
+		assert.ErrorIs(t, err, assert.AnError)
+	})
+	t.Run("fail skips sending for not the first attempt", func(t *testing.T) {
+		poller, _, _, _ := buildWorkflowTaskPoller(t)
+
+		res, err := poller.RespondTaskCompleted(nil, assert.AnError, &s.PollForDecisionTaskResponse{
+			Attempt: common.Int64Ptr(1),
+		}, time.Now())
+		assert.NoError(t, err)
+		assert.Nil(t, res)
+	})
+}
+
+func TestRespondTaskCompleted_Unsupported(t *testing.T) {
+	poller, _, _, _ := buildWorkflowTaskPoller(t)
+
+	assert.Panics(t, func() {
+		_, _ = poller.RespondTaskCompleted(assert.AnError, nil, &s.PollForDecisionTaskResponse{}, time.Now())
+	})
+}
+
+func TestProcessTask_failures(t *testing.T) {
+	t.Run("shutdown", func(t *testing.T) {
+		poller, _, _, _ := buildWorkflowTaskPoller(t)
+		ch := make(chan struct{})
+		poller.shutdownC = ch
+		close(ch)
+
+		err := poller.ProcessTask(&workflowTask{})
+		assert.ErrorIs(t, err, errShutdown)
+	})
+	t.Run("unsupported task type", func(t *testing.T) {
+		poller, _, _, _ := buildWorkflowTaskPoller(t)
+		assert.Panics(t, func() {
```

Comment:
same `PanicsWithValue` thing.

if it's too hard to get equality / feels too over-precise:
```go
assert.panics(t, func() {
  defer func() {
    assert.Contains(t, recover(), "something") // sadly args are reversed
  }()
  thing()
})
```
takes care of it too